File: medicine/medicine/src/main/java/com/medicare/medicine/model/Medicine.java
package com.medicare.medicine.model;


import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table
public class Medicine {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Integer id;

    private String name;
    private String brand;
    private Integer quantity;
    private Double price;
    @Column(columnDefinition = "boolean default true")
    private boolean activate;
    private String description;
    @OneToOne
    @JoinColumn(columnDefinition = "category_Id")
    private Category category;

}

File: NmsCinema/src/app/medicine.ts
import { Category } from "./category";

export class Medicine {

    name : string;
    id:string;
    brand:string;
    quantity:number;
    qty:number;
    price:number;
    activate:boolean;
    description:string;
    category:Category;
    sb:number;

    
    constructor(name:string,brand:string,quantity:number,price:number,activate:boolean,description:string,category:Category)
    {


        this.name  = name;
        this.brand = brand;
        this.quantity = quantity;
        this.price = price;
        this.activate = activate;
        this.description = description;
        this.category = category;
    }
}

